pkg/virt-handler/vm.go

```diff
@@ -890,12 +915,7 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 			return err
 		}
 
-		var match = false
-		for _, version := range d.clusterConfig.GetSupportedAgentVersions() {
-			match = match || regexp.MustCompile(version).MatchString(guestInfo.GAVersion)
-		}
-
-		if !match {
+		if !isGuestAgentSupported(vmi, guestInfo.SupportedCommands) {
```

Comment:
One thing we probably need to be aware of, and perhaps think through a bit, is how we'll handle this during updates.

The guestInfo object is retrieved from virt-launcher, so the first time we update KubeVirt's control plan and new virt-handlers roll out, the "guestInfo.SupportedCommands" list will be empty. It won't be until that VMI is either migrated or restarted later on in a new vmi Pod that we'll get the SupportedCommands.

Either way, there will be a period where there's no "SupportedCommands" and that will result in the unsupported agent condition, even when we technically might support the agent just fine. 

Maybe we should have a fall back to the previous behavior in the event that SupportedCommands is empty? thoughts?

Reply:
On the virt-launcher side, the response from the guest agent gets cached. So this is what will be returned when virt-handler asks. Are you sure it will be empty?

Reply:
This probably does need more thinking, but I feel that it's a general problem for all guest agent information, so probably qualifies as a separate issue/PR--especially since this one is somewhat important to get merged thanks to the fact that qemu guest agent v5 is out now, and we're reporting that Fedora guests are incompatible with KubeVirt.

As for reverting to the previous behavior, that's probably not the best strategy--as time goes on we're pretty much guaranteed to report incompatibility anyway. The fact that it kept breaking every couple of months is what prompted the original direction of this PR (to just delete the compatibility check entirely)

Reply:
> As for reverting to the previous behavior, that's probably not the best strategy--as time goes on we're pretty much guaranteed to report incompatibility anyway. The fact that it kept breaking every couple of months is what prompted the original direction of this PR (to just delete the compatibility check entirely)

the issue is we'll be in the same situation where incompatibility is reported if we dont' preserve the previous behavior.

For example,  We'll update virt-handler but the outdated virt-launcher won't be supplying us the SupportedCommands list because that field is new (old virt-launcher won't give it to us), and then virt-handler will mark that the guest agent isn't supported.

So, if the supportedCommands is completely empty, that indicates we should try and observe the previous (flawed) behavior because it's the only thing we'll have to go on.